package.json

```diff
@@ -16,7 +16,8 @@
 	},
 	"scripts": {
 		"test": "xo && ava",
-		"update-builtin-globals": "node scripts/get-builtin-globals.mjs"
+		"update-builtin-globals": "node scripts/get-builtin-globals.mjs",
+		"update-types": "node scripts/generate-types.mjs > index.d.ts"
```

Comment:
This needs to be called in a `prepublish` script too, so that it fail in case the types are out of date.

Reply:
Added a `prepublishOnly`, do you want it to do a diff as well and fail if its diffing? Or eg. `np` is handling that for you? The correct types will be published anyhow, its whether they are ensured to committed to git or not that's the question.

In a way maybe that should run on CI as well?

Reply:
I think the best solution is to gitignore the type file and just generate it on publish and install. So it should use `prepare` instead.